[17.0][MIG] website_sale_product_attribute_value_filter_existing: Migration to 17.0

lef-adhoc opened the pull request.

CarlosRoca13 committed:
[IMP] : black, isort, prettier

CarlosRoca13 committed:
[MIG] website_sale_product_attribute_value_filter_existing: Migration to v13

sbidoul committed:
[FIX] website_sale_product_attribute_value_filter_existing: development_status

claudiagn committed:
Added translation using Weblate (Spanish)

claudiagn committed:
Added translation using Weblate (Catalan)

bosd committed:
Added translation using Weblate (Dutch)

miguel-S73 committed:
[IMP] website_sale_product_attribute_value_filter_existing: black, isort, prettier

miguel-S73 committed:
[MIG] website_sale_product_attribute_value_filter_existing: Migration to 14.0

CarlosRoca13 committed:
[IMP] website_sale_product_attribute_value_filter_existing: black, isort, prettier

CarlosRoca13 committed:
[MIG] website_sale_product_attribute_value_filter_existing: Migration to 15.0

[UPD] Update website_sale_product_attribute_value_filter_existing.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: e-commerce-15.0/e-commerce-15.0-website_sale_product_attribute_value_filter_existing
Translate-URL: https://translation.odoo-community.org/projects/e-commerce-15-0/e-commerce-15-0-website_sale_product_attribute_value_filter_existing/

anothingguy committed:
[16.0][MIG] website_sale_product_attribute_value_filter_existing: Migration to 16.0

Commit:
[UPD] Update website_sale_product_attribute_value_filter_existing.pot

OCA-git-bot committed:
[UPD] README.rst

HekkiMelody committed:
[FIX] website_sale_product_attribute_value_filter_existing: match signature

match the function signature of shop with the original one from
website_sale, to avoid potential issues with other modules overwriting
the function and passing unexpected parameters

OCA-git-bot committed:
website_sale_product_attribute_value_filter_existing 16.0.1.0.1

OCA-git-bot committed:
[UPD] README.rst

josep-tecnativa committed:
[IMP] website_sale_product_attribute_value_filter_existing: test performance improvement
- Switch to setUpClass for avoiding repeat the same setup for each test.
- Include context keys for avoiding mail operations overhead.

pilarvargas-tecnativa committed:
[FIX] website_sale_product_attribute_value_filter_existing: fix test to check the real functionality of the module

TT45590

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

lef-adhoc committed:
[IMP] website_sale_product_attribute_value_filter_existing: adds filter extraction in mobile view

OCA-git-bot committed:
[BOT] post-merge updates

lef-adhoc committed:
[IMP] website_sale_product_attribute_value_filter_existing: pre-commit auto fixes

lef-adhoc committed:
[MIG] website_sale_product_attribute_value_filter_existing: Migration to 17.0

lef-adhoc commented:
@pilarvargas-tecnativa Hi, may you check this?

pilarvargas-tecnativa commented:
thanks!

@chienandalu can you take a look?

lef-adhoc commented:
@chienandalu Hi, do you have any news about this?

theangryangel commented:
LGTM, functionally tested.

pilarvargas-tecnativa commented:
@lef-adhoc please check pre-commit

theangryangel commented:
Apologies, I have given a positive review, however, it seems that there is a caching issue that we had missed. One of our team is investigating to see if they can assist :)

AndrzejGerasimukARCHIMEDES commented:
@theangryangel 
Hello

Any luck with resolving caching issue ?

Can You share in details what kind of problem did You find?

theangryangel commented:
We ended up needing to go a different direction to meet the customer's need, and I don't think we've come back to look at this. @GarethGloSystems can you remind me what we ran into?

lef-adhoc commented:
@pilarvargas-tecnativa pre-commit is ready

GarethGloSystems commented:
https://github.com/odoo/odoo/commit/29b6b851a64f1f19922c6d0eff01e5327822c21b added `<t t-cache="a,attrib_set">` to the `website_sale.product_attributes` and `website_sale.o_wsale_offcanvas` templates.

The last time I investigated these changes were incompatible with this module.

pedrobaeza commented:
AFAIK, you can remove that cache or expand it with the needed data using the inheritance system (with `replace`, or `t-add`).

pilarvargas-tecnativa committed:
[FIX] website_sale_product_attribute_value_filter_existing: Hide attribute parent container

If only the attribute container is hidden, the parent container still occupies a space that is visible in the attribute list. To avoid this, apply the condition to the parent container and hide it completely.

TT50704

pilarvargas-tecnativa commented:
Please add this fix if necessary https://github.com/OCA/e-commerce/pull/970

pilarvargas-tecnativa reviewed (changes requested):
Please add this fix https://github.com/OCA/e-commerce/pull/970

lef-adhoc commented:
@pilarvargas-tecnativa hi, can you check if everything is ok?

pilarvargas-tecnativa commented:
> @pilarvargas-tecnativa hi, can you check if everything is ok?

@lef-adhoc Hi, but I see you have included the change in your migration commit, you should cherry-pick the commit with the fix to keep the history.

pilarvargas-tecnativa reviewed (approved):
thanks

lef-adhoc commented:
@pedrobaeza Hi, can we merge this?

chienandalu commented on website_sale_product_attribute_value_filter_existing/__manifest__.py:
```diff
@@ -3,7 +3,7 @@
     "name": "Website Sale Attribute Value Existing",
     "summary": "Allow hide attributes values not used in variants",
-    "version": "16.0.1.1.0",
+    "version": "17.0.1.0.0",
```
This change should be in the migration commit, not in the pre-commit one

lef-adhoc commented:
@chienandalu It is corrected

pilarvargas-tecnativa reviewed (changes requested):
In the mobile menu all attributes are displayed even if only one of them is in use.
![image](https://github.com/user-attachments/assets/698d0ccb-75b7-4d64-a1b9-97990aa9ab50)
For the test I have created the attribute TEST, with the values 1, 2 and 3. I have used the value 2 in a product and in the filter panel it appears correct but in the mobile view panel it does not.

pilarvargas-tecnativa commented:
Take a look: https://github.com/odoo/odoo/blob/d39a8a8bb220e8c06b6118f357a220e80ea77716/addons/website_sale/views/templates.xml#L691

pilarvargas-tecnativa commented on website_sale_product_attribute_value_filter_existing/tests/test_website_sale_product_attribute_value_filter_existing.py:
```diff
@@ -0,0 +1,85 @@
+        )
+        cls.product_attribute_value_blue = ProductAttributeValue.create(
+            {"name": "Test blue", "attribute_id": cls.product_attribute.id}
+        )
```
The yellow attribute that is tested in the tour that is not present is not really present because it is not defined. That part needs to be corrected and the unused attribute needs to be added.

pilarvargas-tecnativa replied:
https://github.com/OCA/e-commerce/pull/979

pilarvargas-tecnativa replied:
Hello, can you add the correction of the tests I told you about?

pilarvargas-tecnativa committed:
[FIX] website_sale_product_attribute_value_filter_existing: Add attribute ‘Test yellow'

It is being checked that the attribute ‘Test yellow’ is not present in
the filter panel as it is not being used in any product.
Actually, no real check is being made if the attribute is not defined.
It is a fact, if it is not defined, it does not exist and will not be
present at all.

lef-adhoc commented:
@pilarvargas-tecnativa Hi, can you check this ?

pilarvargas-tecnativa reviewed (approved):
LGTM :+1:  thanks!

pilarvargas-tecnativa commented:
cc @tecnativa TT52391

@chienandalu can you take a look?

pilarvargas-tecnativa commented on website_sale_product_attribute_value_filter_existing/views/templates.xml:
```diff
@@ -0,0 +1,35 @@
+    </template>
+    <template id="o_wsale_offcanvas" inherit_id="website_sale.o_wsale_offcanvas">
+        <xpath
+            expr="//div[@t-elif=&quot;a.display_type in ('radio', 'pills', 'select')&quot;]//div[@t-foreach='a.value_ids']"
```
```suggestion
            expr="//div[@t-elif=&quot;a.display_type in ('radio', 'pills', 'select', 'multi')&quot;]//div[@t-foreach='a.value_ids']"
```

lef-adhoc commented:
@pilarvargas-tecnativa that's ready

pilarvargas-tecnativa reviewed (approved):
thanks!

pedrobaeza commented:
/ocabot migration website_sale_product_attribute_value_filter_existing
/ocabot merge nobump
